### ISSUE TITLE
Campaign Enemy Default ROE

### ISSUE THREAD

**Issue by Novaras** [HW-PlayersPatch/Development#182]
edit: Cloaked edited Fear's issue =)

# Problem
In the hw1 campaign, enemy ships start in defensive ROE. This leads to many issues. You can harvest next to the AI, or salvage the AI, and they will not attack back unless the mission script happens to manually tell them to.

# Implementation
To fix we should make all campaign only ships default to offensive ROE. Would need to use props to do this on Taiidan ships for SP only.
Note that GBX didn't rename hw2's Aggressive ROE to Offensive everywhere. So in the .ship “Offensive” does not work. "Aggressive", “Defensive” and "Passive" do work so this is our solution:
NewShipType.defaultROE=“Aggressive”

# Concern
Will this cause additional problems in some missions, like perhaps the enemy attacking you when they shouldn't? Please leverage your play-through experience and comment. The retaliation range on most ships is only about 5k, so its really not a big deal at all imho.

# Example
One of the MANY examples is Galactic Core (hw1rm m12) lazy AI Reported by a guy in disco: https://discordapp.com/channels/128196211969753088/135543171768647680/620366193424990239
Basically the carrier fleet is totally idle unless provoked, and will ignore you completely until then (you can see him mining right near the fleet).

**Comment by cloaked2222** [comment]
This is common throughout the hw1 campaign. I'm considered making all non-kus units default to offensive ROE. Maybe it breaks something, maybe its fine, idk. Lemme know what you think based on your campaign playthroughs.

**Comment by Novaras** [comment]
Is the AI itself a huge pain to work with?

In my playthrough I kinda demolished everything so I'm not sure. I did notice some oddities such idle kadeshi swarmers (I thought they were outta fuel but they just were sitting there actually) so I guess its not uncommon.

I don't think aggro campaign units would break stuff however I'm almost certain there would be unintended consequences.

Even if it takes a while I think it'd be best to just fix these issues correctly over time?

**Comment by Stuart1998** [comment]
Implementing #176 should help with this, but note that the 1c version of this mission featured a second carrier that I believe had more traditional/developed AI.

**Comment by cloaked2222** [comment]
@Stuart1998 I edited the OP to clarify the root cause for everyone. This issue is separate from 176.

**Comment by cloaked2222** [comment]
bb55b1500cfa9dc282d6acf002946f6b8fd8885e

**Comment by cloaked2222** [comment]
I set all enemy ships in the campaign to default to Offensive tactics. Some scripts may alter their tactics from there. This is done in the SP_Enemy_ROE branch:
08292ce23a9609c5a950bcfbe6e8771da4bae5ba

HW1 and HW2 campaigns would need a full play-through test to implement. Otherwise what should be waves of five separate attacks could turn into one onslaught that wipes out the player.

If this is a problem, would could set certain ships or races back to defensive if needed. It may make sense to change all Taiidan/Vaygr ships back to Defensive, if the player is getting overrun in later missions.
